distributed/http/scheduler/prometheus/core.py

```diff
@@ -37,9 +39,17 @@ def collect(self):
             "Number of workers known by scheduler",
             labels=["state"],
         )
-        worker_states.add_metric(["connected"], len(self.server.workers))
-        worker_states.add_metric(["saturated"], len(self.server.saturated))
         worker_states.add_metric(["idle"], len(self.server.idle))
+        worker_states.add_metric(
+            ["partially saturated"],
+            len(self.server.running)
+            - len(self.server.idle)
+            - len(self.server.saturated),
+        )
+        worker_states.add_metric(["saturated"], len(self.server.saturated))
+        worker_states.add_metric(
+            ["paused/retiring"], len(self.server.workers) - len(self.server.running)
```

Comment:
@ntabris are whitespaces and '/' n the labels OK?

Reply:
no, "They must match the regex `[a-zA-Z_][a-zA-Z0-9_]*`" (from https://prometheus.io/docs/concepts/data_model/)

How hard would it be to expose those as separate numbers? (I assume this requires internal changes but have no idea if those are easy/minor changes or highly non-trivial.)

Reply:
I could break them down easily in Prometheus but it would be O(n) to the total number of workers on the cluster, so I'm reluctant to do it.
It's possible to make it O(1) but it would require additional sets in the scheduler which I don't feel would be justified just for this.